internal/controllers/device_data_controller.go

```diff
@@ -349,7 +348,13 @@ func (udc *UserDevicesController) QueryDeviceErrorCodes(c *fiber.Ctx) error {
 		return err
 	}
 
-	q := &models.ErrorCodeQuery{ID: ksuid.New().String(), UserDeviceID: udi, ErrorCodes: req.ErrorCodes, QueryResponse: chtResp}
+	chtJSON, err := json.Marshal(chtResp)
+	if err != nil {
+		logger.Err(err).Interface("requestBody", req).Msg("Error occurred fetching description for error codes")
+		return fiber.NewError(fiber.StatusInternalServerError, "Error occurred fetching description for error codes")
+	}
+
+	q := &models.ErrorCodeQuery{ID: ksuid.New().String(), UserDeviceID: udi, ErrorCodes: req.ErrorCodes, CodesQueryResponse: null.JSONFrom(chtJSON)}
```

Comment:
Do we need both of these database fields?

My guess is we're going to want to spit out code → description into its own table soon enough.